tachyon/zk/plonk/circuit/expressions/expression.h

```diff
@@ -65,6 +65,13 @@ class Expression {
   virtual uint64_t Complexity() const = 0;
   virtual std::string ToString() const = 0;
 
+  bool operator==(const Expression& other) const {
+    return type_ == other.type_;
+  }
+  bool operator!=(const Expression& other) const {
+    return type_ != other.type_;
```

Comment:
```suggestion
    return !operator==(other);
```
I suggest this for consistency, but did you intend this since it's a simple type checking?

Reply:
Yeap, for simpler one, I just use `a != b`